### PR TITLE
fix: update course_run api

### PR DESCRIPTION
**Pull request by mubbsharanwar**
- update course_run api to edit upgrade_deadline date using publisher app.

DISCO-1622

### CODE REVIEW

File: course_discovery/apps/api/v1/views/course_runs.py [hunk 1]
```diff
@@ -179,6 +179,7 @@ def list(self, request, *args, **kwargs):
 
     @classmethod
     def push_to_studio(cls, request, course_run, create=False, old_course_run_key=None):
+        print('studio api link ', course_run.course.partner.studio_url)
```
Comment (waheedahmed) [inline course_discovery/apps/api/v1/views/course_runs.py]: nit: remove this print statement.

File: course_discovery/apps/course_metadata/models.py [hunk 2]
```diff
@@ -1739,12 +1739,14 @@ def get_seat_upgrade_deadline(self, seat_type):
...
 
-    def update_or_create_seat_helper(self, seat_type, prices):
+    def update_or_create_seat_helper(self, seat_type, prices, upgrade_deadline):
         defaults = {
             'upgrade_deadline': self.get_seat_upgrade_deadline(seat_type),
         }
         if seat_type.slug in prices:
             defaults['price'] = prices[seat_type.slug]
+        if upgrade_deadline:
+            defaults['upgrade_deadline'] = upgrade_deadline
```
Comment (waheedahmed) [inline course_discovery/apps/course_metadata/models.py]: We were supposed to expose the `upgrade_deadline_override` in Publisher instead of `upgrade_deadline` field.

**Review by waheedahmed — CHANGES_REQUESTED** [review]
Also, update the Publisher PR.

File: course_discovery/apps/course_metadata/models.py [hunk 3]
```diff
@@ -1751,10 +1751,11 @@ def update_or_create_seat_helper(self, seat_type, prices):
             type=seat_type,
             draft=True,
             defaults=defaults,
+            upgrade_deadline_override=upgrade_deadline_override,
```
Comment (waheedahmed) [inline course_discovery/apps/course_metadata/models.py]: We need to add a check above this otherwise it will set null if the value is not sent in the PATCH request.

Comment (waheedahmed) [reply course_discovery/apps/course_metadata/models.py]: Also, in my testing, it created duplicate seat objects instead of updating the existing ones and also added the upgrade deadline override value to the Audit seat type which shouldn't have happened.

**Review by waheedahmed — CHANGES_REQUESTED** [review]

File: course_discovery/apps/course_metadata/models.py (hunk 3 above)
Comment (waheedahmed) [reply course_discovery/apps/course_metadata/models.py]: Screenshot 2021-09-17 at 7 39 06 PM

File: course_discovery/apps/course_metadata/models.py [hunk 4]
```diff
@@ -1739,13 +1739,16 @@ def get_seat_upgrade_deadline(self, seat_type):
...
 
-    def update_or_create_seat_helper(self, seat_type, prices):
+    def update_or_create_seat_helper(self, seat_type, prices, upgrade_deadline_override):
         defaults = {
             'upgrade_deadline': self.get_seat_upgrade_deadline(seat_type),
         }
         if seat_type.slug in prices:
             defaults['price'] = prices[seat_type.slug]
 
+        if upgrade_deadline_override and seat_type.slug != Seat.AUDIT:
```
Comment (waheedahmed) [inline course_discovery/apps/course_metadata/models.py]: I guess the upgrade deadline is only meant for Verified seat types and we can update this condition like this:
```suggestion
        if upgrade_deadline_override and seat_type.slug == Seat.VERIFIED:
```

**Review by waheedahmed — CHANGES_REQUESTED** [review]
One change otherwise 👍

File: course_discovery/apps/course_metadata/models.py (hunk 4 above)
Comment (mubbsharanwar) [reply course_discovery/apps/course_metadata/models.py]: if upgrade deadline is only meant for Verified seat types then condition should be like 
`if upgrade_deadline_override and seat_type.slug == Seat.VERIFIED:`

**Review by waheedahmed — APPROVED** [review]
